test/test_evaluator.cpp

```diff
@@ -535,4 +538,24 @@ TEST_F(ExprEvalTest, TernaryOps) {
   }
 }
 
+TEST_F(ExprEvalTest, Permute) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* in =
+      TensorViewBuilder().shape({-1, -1, -1, 6}).dtype(DataType::Float).build();
+  fusion.addInput(in);
+  TensorView* out = permute(in, {0, 3, 1, 2});
+  fusion.addOutput(out);
+
+  at::Tensor in_tensor =
+      at::rand({256}).cuda().as_strided({2, 3, 4, 6}, {128, 32, 8, 1});
+
+  ExpressionEvaluator evaluator;
+  evaluator.bind(in, in_tensor);
+  at::Tensor out_tensor = evaluator.evaluate(out).as<at::Tensor>();
+  EXPECT_THAT(out_tensor.sizes(), ElementsAre(2, 6, 3, 4));
+  EXPECT_THAT(out_tensor.strides(), ElementsAre(128, 1, 32, 8));
```

Comment:
Should we do:
```C++
std::vector<int64_t> new2old = {0, 3, 1, 2};
....
TensorView* out = permute(in, new2old);
...
auto expect = in_tensor.permute(new2old);
EXPECT_EQ(out_tensor.sizes(), expect.sizes());
EXPECT_EQ(out_tensor. strides(), expect. strides());
```

Reply:
For the purpose of the mentioned issue, I prefer checking the aliasing effect of `Set.Permute` explicitly. In other words, if (I know it's rare) `at::permute` stopped producing an alias, I want the test to fail so I'm forced to change the alias analysis and/or the expression evaluator.